src/main/java/com/KacFlor/ShopSpring/controller/ProductController.java

```diff
@@ -56,43 +56,85 @@ public ResponseEntity<?> deleteById(@PathVariable("id") Integer id){
 
     @PreAuthorize("hasAuthority('" + Role.Fields.ADMIN + "')")
     @PatchMapping("/{id}")
-    public ResponseEntity<?> updateProduct(@RequestBody NewProduct newProduct, @PathVariable("id") Integer id){
+    public ResponseEntity<?> update(@RequestBody NewProduct newProduct, @PathVariable("id") Integer id){
         productService.updateProduct(newProduct, id);
         return new ResponseEntity<>(HttpStatus.OK);
     }
 
     @PreAuthorize("hasAuthority('" + Role.Fields.ADMIN + "')")
     @PostMapping("/new")
-    public ResponseEntity<?> createNewProduct(@RequestBody NewProduct newProduct){
+    public ResponseEntity<?> create(@RequestBody NewProduct newProduct){
         productService.addNewProduct(newProduct);
         return new ResponseEntity<>(HttpStatus.ACCEPTED);
     }
 
     @PreAuthorize("hasAuthority('" + Role.Fields.ADMIN + "')")
     @PatchMapping("/{PTid}/promotion/{PNid}/add")
-    public ResponseEntity<?> addPromotionToProductById(@PathVariable("PTid") Integer PTid, @PathVariable("PNid") Integer PNid){
+    public ResponseEntity<?> addPromotionById(@PathVariable("PTid") Integer PTid, @PathVariable("PNid") Integer PNid){
         productService.addPromotion(PTid, PNid);
         return new ResponseEntity<>(HttpStatus.OK);
     }
 
     @PreAuthorize("hasAuthority('" + Role.Fields.ADMIN + "')")
-    @PatchMapping("/{PDid}/promotion/{PNid}/remove")
-    public ResponseEntity<?> removePromotionFromProductById(@PathVariable("PDid") Integer PDid, @PathVariable("PNid") Integer PNid){
-        productService.removePromotion(PDid, PNid);
+    @PatchMapping("/{PTid}/promotion/{PNid}/remove")
+    public ResponseEntity<?> removePromotionById(@PathVariable("PTid") Integer PTid, @PathVariable("PNid") Integer PNid){
+        productService.removePromotion(PTid, PNid);
         return new ResponseEntity<>(HttpStatus.OK);
     }
 
     @PreAuthorize("hasAuthority('" + Role.Fields.ADMIN + "')")
     @PatchMapping("/{PTid}/category/{Cid}/add")
-    public ResponseEntity<?> addCategoryToProductById(@PathVariable("PTid") Integer PTid, @PathVariable("Cid") Integer Cid){
+    public ResponseEntity<?> addCategoryById(@PathVariable("PTid") Integer PTid, @PathVariable("Cid") Integer Cid){
         productService.addCategory(PTid, Cid);
         return new ResponseEntity<>(HttpStatus.OK);
     }
 
     @PreAuthorize("hasAuthority('" + Role.Fields.ADMIN + "')")
-    @PatchMapping("/{PDid}/category/{Cid}/remove")
-    public ResponseEntity<?> removeCategoryFromProductById(@PathVariable("PDid") Integer PDid, @PathVariable("Cid") Integer Cid){
-        productService.removeCategory(PDid, Cid);
+    @PatchMapping("/{PTid}/category/{Cid}/remove")
```

Comment:
do not use verbs in url, we talked about it

Reply:
APPLY EVERYWHERE !!!

Reply:
done in Fixes branch